### ISSUE TITLE
404 page button doesn't do anything

### ISSUE THREAD

**Issue by cheddZy** [huyngxyz/Pillarstack#2]
### Issue:
On the 404 page, when you press the "Go back home" button, it won't take you back home.
**Tried on:** Firefox (No notable plugins) & Chrome Dev (No plugins at all).

### Reproduce:
1. Go to [https://www.pillarstack.com/404](https://www.pillarstack.com/404) (or any non-existing path)
2. Press the Go back home button
3. OR press the top right logo button, that acts as a home/back button
4. The URL should now show as the correct "`https://www.pillarstack.com/`" but the content on the page hasn't changed. Upon refreshing the page at that point you'll see the actual home page again.

If you leave the 404 page open for a little bit (it's a bit inconsistent mind you, but I was able to replicate the behavior a few times) then pressing the back button / home button will result in seeing the right content on the page in comparison to immediately pressing it upon visiting the 404 page.

No idea what this could be, I'm not experienced with Next.js routing so would be interested to know & get tagged on what the problem ended up being.

**Comment by huyngxyz** [comment]
@cheddZy Thanks for pointing this out! This is super strange and I don't have any idea why it's not working either haha. I'll dig into it whenever I have time : )

**Comment by Xiaozuoyi** [comment]
```jsx
"use client";
import Button from "@/components/Button/Button";
import { useRouter } from "next/navigation";

export default function ErrorPage() {
    const router = useRouter();
  const goBackHome = () => {
    router.push("/");
  }
  return (
    <div className="h-screen flex items-center justify-center -m-20 flex-col ">
      <Button onClick={goBackHome}>Go Back home</Button>
    </div>
  );
}
```
**When redirecting links, we can use either next/link or next/router to achieve it. Here, we will use next/router for redirection.**

I don't have the environment configuration file .env, so the webpage cannot run. I cannot determine if the modification is correct.

**Comment by gursewak8642** [comment]
![Screenshot 2024-03-05 225358](https://github.com/huyngxyz/Pillarstack/assets/82928107/2ebf1eaf-63c8-4345-9743-187cbec6e663)

I am getting this error while settling up the project what should i do ?

**Comment by 7D-exe** [comment]
I tested @Xiaozuoyi fix and **it worked**, I replaced the getContent function with this code below so that I can see the webpage
```// /app/utils/getContent.jsx

const localData = {
  items: [
  ],
  total: 0, 
};

export async function getContent({ tag, category, page }) {
  
  return {
    items: localData.items, 
    total: localData.total,
  };
}

```
I wanted to do a pull request but I didn't fix that so test it yourself with the new function that gives demo data :)

**Comment by gursewak8642** [comment]
@7D-exe  I am trying to setup the repo but it is asking for the tocken acess and acess_key as i have shared the screenshot above. Could you please share the .env setup file or help me to resolve the above error.

**Comment by 7D-exe** [comment]
@gursewak8642 replace getContent func with the code I provided just copy and paste it in the same path written in the top of the snippet, you don't need to create .env file if you replaced the func

**Comment by gursewak8642** [comment]
@7D-exe Yeah its working fine but none of the tool's and suggestions are visible 
![image](https://github.com/huyngxyz/Pillarstack/assets/82928107/3f7c5a00-4714-4a36-b4e9-b6fa98567b92)

**Comment by 7D-exe** [comment]
@gursewak8642 that's because the site is built in a way to get tools/sites via an API, you need the dev keys to get the same sites in ur end, but you work on the site now.

its built like that so when he wants to upload a new site/tool he upload it in the database(Contentful) instead of updating the site code manually each time

**Comment by huyngxyz** [comment]
> ```js
> "use client";
> import Button from "@/components/Button/Button";
> import { useRouter } from "next/navigation";
> 
> export default function ErrorPage() {
>     const router = useRouter();
>   const goBackHome = () => {
>     router.push("/");
>   }
>   return (
>     <div className="h-screen flex items-center justify-center -m-20 flex-col ">
>       <Button onClick={goBackHome}>Go Back home</Button>
>     </div>
>   );
> }
> ```
> 
> **When redirecting links, we can use either next/link or next/router to achieve it. Here, we will use next/router for redirection.**
> 
> I don't have the environment configuration file .env, so the webpage cannot run. I cannot determine if the modification is correct.

Awesome, I also had the same idea using router.push. I'll be giving that a try!

**Comment by huyngxyz** [comment]
Also, bear with me since it's my first open source project haha but thanks for reporting these issues. I'll try to update the ReadME file with a proper project setup guide so you guys can get it up and running without any problems : )

**Comment by gursewak8642** [comment]
@huyngxyz Yes, sir. Of course, hats off to your efforts 🙌🙌, as I got to know about this from your Youtube channel. As I shared PillarStack with my friends, they also liked it a lot. One more thing could you please settle the components of pillerStack so that we can also access it in our local host too 🙌.

**Comment by Xiaozuoyi** [comment]
I found the reason. It was passed in the wrong way: href="/" ==> herf={"/"}
[CleanShot 2024-03-14 at 13 00 25@2x](https://github.com/huyngxyz/Pillarstack/assets/94602408/3ce471b3-a7ed-4bb7-b40a-9023823487be)
